src/CMake/FindVisItPython.cmake

```diff
@@ -150,45 +148,105 @@ if(PYTHONINTERP_FOUND)
             set(PYTHON_USE_UNDEFINED_DYNAMIC_LOOKUP_FLAG OFF)
         endif()
 
+        # our goal is to find the specific python lib, based on info
+        # we extract from distutils.sysconfig from the python executable
+        #
         # check for python libs differs for windows python installs
         if(NOT WIN32)
             # we may build a shared python module against a static python
             # check for both shared and static libs cases
 
-            # check for shared first
-            set(PYTHON_GLOB_TEST "${PYTHON_LIB_DIR}/libpython*${CMAKE_SHARED_LIBRARY_SUFFIX}")
-            FILE(GLOB PYTHON_GLOB_RESULT ${PYTHON_GLOB_TEST})
-            # then for static if shared is not found
-            if(NOT PYTHON_GLOB_RESULT)
-                set(PYTHON_GLOB_TEST "${PYTHON_LIB_DIR}/libpython*${CMAKE_STATIC_LIBRARY_SUFFIX}")
+            # combos to try:
+            # shared:
+            #  LIBDIR + LDLIBRARY
+            #  LIBPL + LDLIBRARY
+            # static:
+            #  LIBDIR + LIBRARY
+            #  LIBPL + LIBRARY
+
+            execute_process(COMMAND "${PYTHON_EXECUTABLE}" "-c" 
+                                    "import sys;from distutils.sysconfig import get_config_var; sys.stdout.write(get_config_var('LIBDIR'))"
+                            OUTPUT_VARIABLE PYTHON_CONFIG_LIBDIR
+                            ERROR_VARIABLE  ERROR_FINDING_PYTHON_LIBDIR)
+
+            execute_process(COMMAND "${PYTHON_EXECUTABLE}" "-c" 
+                                    "import sys;from distutils.sysconfig import get_config_var; sys.stdout.write(get_config_var('LIBPL'))"
+                            OUTPUT_VARIABLE PYTHON_CONFIG_LIBPL
+                            ERROR_VARIABLE  ERROR_FINDING_PYTHON_LIBPL)
+
+            execute_process(COMMAND "${PYTHON_EXECUTABLE}" "-c" 
+                                    "import sys;from distutils.sysconfig import get_config_var; sys.stdout.write(get_config_var('LDLIBRARY'))"
+                            OUTPUT_VARIABLE PYTHON_CONFIG_LDLIBRARY
+                            ERROR_VARIABLE  ERROR_FINDING_PYTHON_LDLIBRARY)
+
+            execute_process(COMMAND "${PYTHON_EXECUTABLE}" "-c" 
+                                    "import sys;from distutils.sysconfig import get_config_var; sys.stdout.write(get_config_var('LIBRARY'))"
+                            OUTPUT_VARIABLE PYTHON_CONFIG_LIBRARY
+                            ERROR_VARIABLE  ERROR_FINDING_PYTHON_LIBRARY)
+
+            message(STATUS "PYTHON_CONFIG_LIBDIR:     ${PYTHON_CONFIG_LIBDIR}")
+            message(STATUS "PYTHON_CONFIG_LIBPL:      ${PYTHON_CONFIG_LIBPL}")
+            message(STATUS "PYTHON_CONFIG_LDLIBRARY:  ${PYTHON_CONFIG_LDLIBRARY}")
+            message(STATUS "PYTHON_CONFIG_LIBRARY:    ${PYTHON_CONFIG_LIBRARY}")
+
+            set(PYTHON_LIBRARY "")
+            # look for shared libs first
+            # shared libdir + ldlibrary
```

Comment:
Question: If we are building a static version of VisIt, do we still want to find python shared libs?

Reply:
good point, however when we build static with build visit, would we expect to the shared python libs to exist? 

system install it could happen for sure.